File: Selenium/Project/Activity8.py
# Import required packages from selenium
import time

import pytest
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait


class TestClass():

    def test_logins(self, my_fixture):
        # Initialise Driver
        driver = my_fixture
        url = 'https://alchemy.hguy.co/jobs/wp-admin'
        driver.get(url)

        # Initialise Wait for 10 secs
        wait = WebDriverWait(driver, 10)

        userTextBox = driver.find_element(By.ID, "user_login")
        userPassBox = driver.find_element(By.ID, "user_pass")
        submitButton = driver.find_element(By.ID, "wp-submit")

        userTextBox.send_keys("root")
        userPassBox.send_keys("pa$$w0rd")
        submitButton.click()

        loginMsg = driver.find_element(By.XPATH, "//li[@id='wp-admin-bar-my-account']/a/span")
        assert "root" == loginMsg.text

        time.sleep(3)

File: Selenium/Project/Activity8.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.testng.annotations.AfterClass;
import org.junit.jupiter.api.Assertions;

import java.time.Duration;
import java.util.Set;

public class Activity8 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setup() {
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs/wp-admin");
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    @Test
    public void verifyLogin() throws InterruptedException {

        WebElement userTextBox = driver.findElement(By.id("user_login"));
        WebElement userPassBox = driver.findElement(By.id("user_pass"));
        WebElement submitButton = driver.findElement(By.id("wp-submit"));

        userTextBox.sendKeys("root");
        userPassBox.sendKeys("pa$$w0rd");
        submitButton.click();

        WebElement loginMsg = driver.findElement(By.xpath("//li[@id='wp-admin-bar-my-account']/a/span"));
        Assertions.assertEquals("root", loginMsg.getText());

    }

    @AfterClass
    public void cleanup() {
        driver.quit();
    }
}
